### PR TITLE
feat: reorganize README documentation and expand documentation

### PR DESCRIPTION
**Pull request by miguelsorianod**
## Description

Several changes to the documentation.

Simplify the main README documentation. The main README had grown too much losing part of its usefulness. The idea is that the main README provides the most useful information first and links to other relevant documents where more detail is provided in case the user is interested.

Some additional documents have been created, being of special relevance the populating configuration document and the interacting with fleet manager document.

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

**Comment by miguelsorianod** [comment]
Aside from the diff I recommend viewing the files in full view

File: docs/interacting-with-fleet-manager.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+## Interacting using the RHOAS CLI
+
+An alternative to interact with the Fleet Manager API is to make use of the
+[RHOAS CLI](https://github.com/redhat-developer/app-services-cli).
+
+First [run the Fleet Manager locally](../README.md#running-fleet-manager-for-the-first-time-in-your-local-environment)
+
+Then, run the RHOAS CLI pointing to the locally running Fleet Manager:
+```
+./rhoas login --mas-auth-url=stage --api-gateway=http://localhost:8000
```
Comment (machi1990) [inline docs/interacting-with-fleet-manager.md:92]: the `mas-auth-url` flag has been dropped from recent versions of the CLI

File: docs/interacting-with-fleet-manager.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+> Note: The locally installed kas-fleet-manager doesn't deploy TLS enabled
+kafka admin server but the default URL scheme used by the app-service-cli is
+HTTPS. So, the url scheme should be changed to http in the
```
Comment (machi1990) [inline docs/interacting-with-fleet-manager.md:97]: It should be possible to configure the locally KFM to have TLS enabled for the admin server. 
There are the `enable-cname-registration` and `enable-kafka-external-listener` flags.

File: docs/observability/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+The file [metrics.go](../../pkg/metrics/metrics.go) creates Prometheus metrics
+of differing types. These are the metrics which are then reported and visualised
+in each Grafana dashboard.
+See [here](https://prometheus.io/docs/concepts/metric_types/) for more info
+about Prometheus metric types.
+
+These metrics are grouped by metric subject: data plane clusters,
+service ('kafka') and reconcilers.
+
+There is also a section of metrics regarding the Observatorium service. See [kas-fleet-manager metrics configmap](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/resources/observability/grafana/grafana-dashboard-kas-fleet-manager-metrics.configmap.yaml#L5460-7247)
```
Comment (machi1990) [inline docs/observability/README.md:14]: are the links still valid?
we moved out of app-interface IIRC

File: docs/observability/README.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+with Observatorium to retrieve the metrics sent by the managed
+service from the data plane.
+
+### Configuring Observatorium
+
+To configure a new managed service to use Observatorium a Red Hat managed
+Observatorium service can be used. For that, a new `Observatorium Tenant` has
+to be created and configured in that Red Hat managed Observatorium service. That
+task is done by the Red Hat Observability team. To do so there's
+an Onboarding process. See the[Onboarding a Tenant into Red Hat’s Observatorium Instance](https://docs.google.com/document/d/1pjM9RRvij-IgwqQMt5q798B_4k4A9Y16uT2oV9sxN3g) document on how to do it.
```
Comment (machi1990) [inline docs/observability/README.md:52]: Is this relevant to an already deployed service?

File: docs/interacting-with-fleet-manager.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+```
+curl -v -X DELETE -H "Authorization: Bearer $(ocm token)" http://localhost:8000/api/kafkas_mgmt/v1/kafkas/<kafka_request_id>?async=true
+```
+
+### Using the Kafka Admin Server API
+
+The Kafka Admin Server API is used for managing topics, acls, and consumer groups
+on a given Kafka Request.
+
+The Kafka Admin Server OpenAPI specification can be found [here](https://github.com/bf2fc6cc711aee1a0c2a/kafka-admin-api/blob/main/kafka-admin/src/main/resources/openapi-specs/kafka-admin-rest.yaml).
```
Comment (machi1990) [inline docs/interacting-with-fleet-manager.md:72]:
```suggestion
The Kafka Admin Server OpenAPI specification can be found [here](https://github.com/bf2fc6cc711aee1a0c2a/kafka-admin-api/blob/main/kafka-admin/.openapi/kafka-admin-rest.yaml).
```

the link doesn't exist. I am suggesting a different one but would need @MikeEdgar to confirm if it is the current used openapi file for the admin server.

File: Makefile [hunk 6]
```diff
@@ -527,6 +529,33 @@ cos-fleet-catalog-camel/teardown:
 	rm config/connector-types/cos-fleet-catalog-camel
 .PHONY: cos-fleet-catalog-camel/teardown
 
+# Touch all the necessary files for fleet manager to start
+# See docs/populating-configuration.md for more information
+secrets/setup/empty:
```
Comment (machi1990) [inline Makefile:534]: Nice. I am wondering if we can document somewhere (e.g https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/tree/aa91c8702ad01daa7da5840f792de435bcd04bef/docs/best-practices) that this list needs to be kept up to date

File: docs/interacting-with-fleet-manager.md (hunk 5 above)
Comment (machi1990) [reply docs/interacting-with-fleet-manager.md:72]: thanks @MikeEdgar

**Comment by machi1990** [comment]
Thanks @miguelsorianod left some suggestions.

The propose restructure looks good to me. I'll let others chime in as well.

File: docs/observability/README.md (hunk 3 above)
Comment (miguelsorianod) [reply docs/observability/README.md]: That's correct. I removed this part as it does not apply anymore

File: docs/interacting-with-fleet-manager.md (hunk 2 above)
Comment (miguelsorianod) [reply docs/interacting-with-fleet-manager.md]: :+1: Note deleted.

File: docs/interacting-with-fleet-manager.md (hunk 1 above)
Comment (miguelsorianod) [reply docs/interacting-with-fleet-manager.md]: :+1: updated to the currently used flag

File: docs/interacting-with-fleet-manager.md (hunk 5 above)
Comment (miguelsorianod) [reply docs/interacting-with-fleet-manager.md]: updated :+1:

File: docs/getting-credentials-and-accounts.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+# User Account & Organization Setup
+1. Create new organization and users being members of this organization:
+* Make sure that you are logged out from any existing `access.redhat.com` sessions. Use the `register new organization` flow to create an organization
+* Create new users using user management UI. It is recommended to append or prefix usernames of your team members, e.g. `dffrench_control_plane` for `dffrench` username being a member of `Control Plane` team.
+
+2. To onboard a fleet-manager to [app-interface](https://gitlab.cee.redhat.com/service/app-interface) a new role will be required for resources (e.g. Syncsets, etc) creation (see [example](https://gitlab.cee.redhat.com/service/uhc-account-manager/-/blob/master/pkg/api/roles/managed_kafka_service.go)). The following [MR](https://gitlab.cee.redhat.com/service/uhc-account-manager/-/merge_requests/1907) shows how the ManagedKafkaService role was created for RHOSAK.
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply docs/getting-credentials-and-accounts.md]: Reworded

File: Makefile (hunk 6 above)
Comment (miguelsorianod) [reply Makefile:534]: I will think about this and maybe add some information if I end up finding it appropriate in a potential future PR.
At the moment I don't have a clear idea or where this would be or if would even add any substantial as the same thinking can be applied to lots of tasks related to Makefile/templates

File: docs/observability/README.md (hunk 4 above)
Comment (miguelsorianod) [reply docs/observability/README.md:51]: This repository doesn't assume RHOSAK as the only audience.
Not all users potential users/developers of KFM are RHOSAK service

File: README.md [hunk 8]
```diff
@@ -16,275 +16,78 @@ works, see the [implementation documentation](docs/implementation.md).
...
-   scopes/permissions list (which will check each of the subcategory
-   boxes beneath it)
-1. Copy the value of your Personal Access Token to a secure private location.
-   Once you leave the page, you cannot access the value again & you will be
-   forced to reset the token to receive a new value should you lose the original
-1. Paste the token value in the `secrets/observability-config-access.token`
-   file
-
-### Data Plane OSD cluster setup
+There are a some additional prerequisites required for running kas-fleet-manager
```
Comment (vmanley) [inline README.md:19]:
```suggestion
There are some additional prerequisites required for running kas-fleet-manager
```

File: docs/getting-credentials-and-accounts.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+> NOTE
+Within the AWS account used to provision OSD clusters, you must create an osdCcsAdmin IAM user with the following requirements:
+- This user needs at least Programmatic access enabled.
+- This user must have the AdministratorAccess policy attached to it.
+See https://docs.openshift.com/dedicated/osd_planning/aws-ccs.html#ccs-aws-customer-procedure_aws-ccs for more information.
+
+# SSO authentication
+Depending on whether interacting with public or private endpoints, the authentication can be set up as follows:
+* Red Hat SSO sso.redhat.com for public endpoints
+* Red HAT SSO or MAS SSO for private endpoints. Ssee this [doc](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/docs/mas-sso/service-architecture/service-architecture.md) to understand how kas-fleet-manager is using MAS-SSO for authentication
```
Comment (vmanley) [inline docs/getting-credentials-and-accounts.md:45]:
```suggestion
* Red HAT SSO or MAS SSO for private endpoints. See this [doc](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/docs/mas-sso/service-architecture/service-architecture.md) to understand how kas-fleet-manager is using MAS-SSO for authentication
```

**Comment by vmanley** [comment]
Good improvement. Thanks Miguel. Just a couple of typos for me

File: docs/observability/README.md (hunk 4 above)
Comment (machi1990) [reply docs/observability/README.md:51]: The intent is then to have this section so that other services can know how to request a tenant to Observatorium?

File: Makefile (hunk 6 above)
Comment (machi1990) [reply Makefile:534]: Okay sounds good to me. 

PR template could be another place to add it as well - but I am happy to do it once we are clear of the place we want to add it.

File: docs/observability/README.md (hunk 4 above)
Comment (miguelsorianod) [reply docs/observability/README.md:51]: Not other services, but other users that want to deploy their own KFM

File: docs/getting-credentials-and-accounts.md (hunk 9 above)
Comment (miguelsorianod) [reply docs/getting-credentials-and-accounts.md:45]: Fixed :+1:

File: docs/observability/README.md (hunk 4 above)
Comment (machi1990) [reply docs/observability/README.md:51]: I think I get the intent of what you are describing now. I'd suggest to reword the section in a way that it clearly reflects that intent.
For example, 

`"To configure a new managed service to use Observatorium a Red Hat managed
Observatorium service can be used. For that, a new `Observatorium Tenant` has
to be created and configured in that Red Hat managed Observatorium service...."` 

can be rephrased to (just a suggestion) 

`"To configure kas-fleet-manager to use Red Hat managed Observatorium, a tenant & service account credentals (separate from RHOSAK's Stage&Prod ones) have to be created in Red Hat managed Observatorium. That
task is done by the Red Hat Observability team via an onboarding process explained in the[Onboarding a Tenant into Red Hat’s Observatorium Instance](https://docs.google.com/document/d/1pjM9RRvij-IgwqQMt5q798B_4k4A9Y16uT2oV9sxN3g) document."`

File: docs/populating-configuration.md [hunk 10]
```diff
@@ -0,0 +1,353 @@
+> NOTE some step in this document might refer to Red Hat internal components
```
Comment (JameelB) [inline docs/populating-configuration.md:1]:
```suggestion
> NOTE: some step in this document might refer to Red Hat internal components
```

File: docs/populating-configuration.md [hunk 11]
```diff
@@ -0,0 +1,353 @@
...
+  * There is a limit of one instance per user
+* Standard instances
+  * Instances of this type are not automatically deleted
+  * In order to be able to create an instance of this type, the user that
+    is creating it must have enough quota
+
+If you are not interested in making use of Standard Kafka instances you
+can skip this section. Otherwise, keep reading below.
+
+As commented above, In order o be able to create an instance of this type, the
```
Comment (JameelB) [inline docs/populating-configuration.md:60]:
```suggestion
As commented above, in order to be able to create an instance of this type, the
```

File: docs/populating-configuration.md [hunk 12]
```diff
@@ -0,0 +1,353 @@
...
+
+## Allowing creation of *Standard* Kafka instances
+
+Fleet Manager is able to create two types of Kafka instances:
+* Developer instances
+  * Instances of this type are automatically deleted after 48 hours by default
+    > NOTE: This can be controlled by setting the `lifespanSeconds` attribute
+      in the `config/kafka-instance-types-configuration.yaml` configuration file
+      for developer instances
+  * All authenticated users that make use of the Fleet Manager can
```
Comment (JameelB) [inline docs/populating-configuration.md:49]: This is not entirely true. You can only create a developer instance if your account does not have any other rhosak quota allowance. Maybe we can rephrase this to make that clearer?

File: docs/populating-configuration.md [hunk 13]
```diff
@@ -0,0 +1,353 @@
...
+    request the creation of a Kafka developer instance
+  * There is a limit of one instance per user
```
Comment (JameelB) [inline docs/populating-configuration.md:51]: note that this is by default and can actually be changed via configuration: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/internal/environments/development.go#L40

File: docs/observability/README.md (hunk 4 above)
Comment (miguelsorianod) [reply docs/observability/README.md:51]: That makes sense :+1: . Updated

File: docs/populating-configuration.md [hunk 14]
```diff
@@ -0,0 +1,353 @@
...
+Finally, make sure that `gcp` is listed as a supported Cloud Provider with at
+least one configured [GCP region](https://cloud.google.com/compute/docs/regions-zones)
+in Fleet Manager in the `config/provider-configuration.yaml` configuration
+file. See the documentation in that file for detail on the configuration
+schema.
+
+## Setup additional SSO configuration
+A SSO server is needed in Fleet Manager to enable some functionalities:
+* To enable communication between the Fleetshard operator (in the data plane) and the Fleet Manager
+* To configure the OpenShift cluster identity provider in Data Plane clusters
```
Comment (JameelB) [inline docs/populating-configuration.md:159]: Another purpose of it is to allow a successful Kafka instance provision. Each Kafka instance requires an SSO service account for the Canary deployment.

File: docs/populating-configuration.md [hunk 15]
```diff
@@ -0,0 +1,353 @@
...
+
+To configure the Fleet Manager with this authenticated registry information so
+the previously mentioned Data Plane elements can pull container images from it:
+* Base-64 encode your [Docker configuration file](https://docs.docker.com/engine/reference/commandline/cli/#docker-cli-configuration-file-configjson-properties).
+* Copy the contents generated from the previous point into the `secrets/image-pull.dockerconfigjson` file
+
+## Setup the Observability stack secrets
+Fleet Manager provides the ability to provide and configure monitoring metrics
+related to the Data Planes and its Kafka instances. To do so it makes
+use of what's called the ([Observability Stack](./observability/README.md#observability-stack)
```
Comment (JameelB) [inline docs/populating-configuration.md:235]:
```suggestion
use of what's called the ([Observability Stack](./observability/README.md#observability-stack))
```

File: docs/getting-credentials-and-accounts.md [hunk 16]
```diff
@@ -0,0 +1,55 @@
+> NOTE this document refers to Red Hat internal components
```
Comment (JameelB) [inline docs/getting-credentials-and-accounts.md:1]:
```suggestion
> NOTE: this document refers to Red Hat internal components
```

File: docs/getting-credentials-and-accounts.md [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+# User Account & Organization Setup
+
+1. Create new Red Hat organization and users belonging to this organization:
+* Make sure that you are logged out from any existing `access.redhat.com` sessions. Use the `register new organization` flow to create an organization
+* Create new users using user management UI. It is recommended to append or prefix usernames of your team members, e.g. `dffrench_control_plane` for `dffrench` username being a member of `Control Plane` team.
+2. An AMS Role defined through the OCM Resources mechanism has been defined, available for developers of the KAS Fleet Manager to get the needed permissions
+   to perform KAS Fleet Manager development. The definition of the role can be found in 
+   https://gitlab.cee.redhat.com/service/uhc-account-manager/-/blob/master/pkg/api/roles/managed_kafka_service.go.
+3. Users interested in developing KAS Fleet Manager should have this role assigned to their OCM account. This can be done using the OCM Resources
+   mechanism. An example of assigning the AMS role previously mentioned to a user can be found [here](https://gitlab.cee.redhat.com/service/ocm-resources/-/merge_requests/812).
```
Comment (JameelB) [inline docs/getting-credentials-and-accounts.md:16]: This MR may not be the best example for it as the scope definition needed for this has changed i believe. I think the scope needs to be empty now to access some resources in ams i.e. https://gitlab.cee.redhat.com/service/ocm-resources/-/blob/master/data/uhc-stage/users/ppaszki_kafka_service.yaml#L12

See [MR](https://gitlab.cee.redhat.com/service/ocm-resources/-/merge_requests/1527) where this was removed.

File: docs/getting-credentials-and-accounts.md [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+4. KAS Fleet Manager developers also need AMS quota for installing specific OCM add-ons that KAS Fleet Manager requires. An example showing the required quotas for the addons can be found [here](https://gitlab.cee.redhat.com/service/ocm-resources/-/blob/master/data/uhc-stage/orgs/13640203.yaml).
+
+# Vault
+For automation (fleet-manager deployments and app-interface pipelines) it is required to use [vault](https://gitlab.cee.redhat.com/service/app-interface#manage-secrets-via-app-interface-openshiftnamespace-1yml-using-vault) for secrets management
+
+# AWS
+AWS accounts are required for provisioning development OSD clusters and to access AWS Route53 for Domain Name System. To request a AWS accounts, have a chat with your team's manager.
+> NOTE, if you are in the Middleware, make sure to send an email to `mw-billing-leaders@redhat.com` request the account to be created. The email should follow the below format:
```
Comment (JameelB) [inline docs/getting-credentials-and-accounts.md:24]: are we ok to put this mailing list publicly?

File: docs/getting-credentials-and-accounts.md [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+Depending on whether interacting with public or private endpoints, the authentication can be set up as follows:
+* Red Hat SSO sso.redhat.com for public endpoints
+* Red HAT SSO or MAS SSO for private endpoints. See this
+[doc](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/docs/mas-sso/service-architecture/service-architecture.md)
+to understand how kas-fleet-manager is using MAS-SSO for authentication
+
+See [feature-flags](feature-flags.md#Keycloak) to understand flags used for authentication.
+
+## sso.redhat.com service account
+To avail of all required OCM services the fleet-manager depends on, it is
```
Comment (JameelB) [inline docs/getting-credentials-and-accounts.md:52]: Let's note that this isn't required as people can use their own offline tokens for development. 

This is only necessary if it's required to have a dedicated service account for running KFM.

File: docs/observability/README.md [hunk 20]
```diff
@@ -0,0 +1,80 @@
...
+uses.
+
+See the [JSON Model](https://grafana.com/docs/grafana/latest/dashboards/json-model/?pg=docs)
+section for more info about Grafana Dashboard JSON models
+
+## Further information
+See [SLOs README](../slos/README.md) for more informtion about metrics and
+their use in measuring SLIs.
+
+> NOTE this document contains references to Red Hat internal components
```
Comment (JameelB) [inline docs/observability/README.md:32]:
```suggestion
> NOTE: this document contains references to Red Hat internal components
```

File: docs/getting-credentials-and-accounts.md (hunk 18 above)
Comment (miguelsorianod) [reply docs/getting-credentials-and-accounts.md:24]: This change is not adding anything new, it was already public. Specifically, it was introduced in commit `d8a0e40c3e1082802261a62d8a251c79570946d0` in the README.md, and later reorganized on other docs.

Comment (miguelsorianod) [reply docs/getting-credentials-and-accounts.md]: updated

File: docs/populating-configuration.md [hunk 21]
```diff
@@ -0,0 +1,353 @@
...
+populating its configurations.
+
+For convenience, the following Makefile target is provided:
+
+```
+make secrets/setup/empty
+```
+
+This target initializes empty files that are necessary for the service to boot.
+It can be used to quickly get the service running locally, although it does not
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply docs/populating-configuration.md:15]: This change does not seem correct to me. With the new suggested change would be:
`It can be used to quickly to get ...`.

It should either be `It can be used to quickly get ...` or `It can be used quickly to get the ... `

File: docs/populating-configuration.md (hunk 11 above)
Comment (miguelsorianod) [reply docs/populating-configuration.md]: updated :+1:

File: docs/populating-configuration.md (hunk 15 above)
Comment (miguelsorianod) [reply docs/populating-configuration.md]: The suggested change is not correct. The correct change is removing the initial parentheses.

I updated the text to do so.

File: docs/getting-credentials-and-accounts.md (hunk 16 above)
Comment (miguelsorianod) [reply docs/getting-credentials-and-accounts.md]: updated :+1

File: README.md (hunk 8 above)
Comment (miguelsorianod) [reply README.md]: updated :+1:

File: docs/observability/README.md (hunk 20 above)
Comment (miguelsorianod) [reply docs/observability/README.md]: updated :+1:

File: docs/populating-configuration.md (hunk 21 above)
[the remark this answers is not in the document]
Comment (JameelB) [reply docs/populating-configuration.md:15]: misread this, didn't see the `to` at the start. can ignore this one 👍

File: docs/populating-configuration.md [hunk 22]
```diff
@@ -0,0 +1,353 @@
...
+Follow the subsections to find information about how to configure these files properly.
+
+## Interacting with the Fleet Manager API
+
+See the [Interacting with Fleet Manager](interacting-with-fleet-manager.md)
+document.
+
+## Setting up OCM tokens
+
+The Fleet Manager itself requires the use of an OCM token so it can
```
[the remark this answers is not in the document]
Comment (JameelB) [reply docs/populating-configuration.md:27]: Yes, it would work with the temporary, non-offline tokens. But as you said, that token expires after 7-10m so it's not really usable when running kfm as you've to update it every few mins. It would be best to just ensure everyone is using their offline token here to avoid it. 

> Am I correct or I am confusing it with another scenario?

You're correct but you may also be remembering that offline tokens cannot be used for interacting with the kfm api. It has to be the generated non-offline token as the offline token don't have all the claims kfm needs.

File: Makefile [hunk 23]
```diff
@@ -527,6 +529,33 @@ cos-fleet-catalog-camel/teardown:
...
+	touch ${SECRETS_DIR}/ocm-service.clientId
+	touch $(SECRETS_DIR)/ocm-service.clientSecret
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply Makefile:531]: Added

File: docs/populating-configuration.md (hunk 12 above)
Comment (miguelsorianod) [reply docs/populating-configuration.md]: rephrased

File: docs/populating-configuration.md (hunk 13 above)
Comment (miguelsorianod) [reply docs/populating-configuration.md:67]: reworded

File: docs/populating-configuration.md [hunk 24]
```diff
@@ -0,0 +1,353 @@
...
+provide a functional kas-fleet-manager.
+
+Follow the subsections to find information about how to configure these files properly.
+
+## Interacting with the Fleet Manager API
+
+See the [Interacting with Fleet Manager](interacting-with-fleet-manager.md)
+document.
+
+## Setting up OCM tokens
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply docs/populating-configuration.md]: I modified this section

File: docs/getting-credentials-and-accounts.md (hunk 17 above)
Comment (miguelsorianod) [reply docs/getting-credentials-and-accounts.md]: updated

File: docs/populating-configuration.md (hunk 14 above)
Comment (miguelsorianod) [reply docs/populating-configuration.md:177]: :+1: but I understand that is not strictly required.
We have a flag EnableAuthenticationOnKafka that I understand controls that requirement

Comment (miguelsorianod) [reply docs/populating-configuration.md:177]: Added

File: docs/populating-configuration.md (hunk 22 above)
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply docs/populating-configuration.md]: > You're correct but you may also be remembering that offline tokens cannot be used for interacting with the kfm api. It has to be the generated non-offline token as the offline token don't have all the claims kfm needs.

This case is already documented properly in the "Interacting with Fleet Manager" document

[the remark this answers is not in the document]
Comment (miguelsorianod) [reply docs/populating-configuration.md]: > Yes, it would work with the temporary, non-offline tokens. But as you said, that token expires after 7-10m so it's not really usable when running kfm as you've to update it every few mins. It would be best to just ensure everyone is using their offline token here to avoid it.

That seems reasonable. I updated this section to refer to it as an OCM offline token

File: docs/populating-configuration.md [hunk 25]
```diff
@@ -0,0 +1,353 @@
...
+interact with OCM to perform management of Data Plane clusters.
+
+In order for the Fleet Manager to do so, an OCM offline token should be configured.
+To do so, retrieve your OCM offline token and then configure it for Fleet
+Manager by running:
+```
+make ocm/setup OCM_OFFLINE_TOKEN=<your-retrieved-ocm-offline-token>
+```
+
+> NOTE: Information about how to retrieve an OCM token can be found in the
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply docs/populating-configuration.md:53]: updated

File: docs/getting-credentials-and-accounts.md (hunk 19 above)
Comment (miguelsorianod) [reply docs/getting-credentials-and-accounts.md]: Updated

File: docs/populating-configuration.md [hunk 26]
```diff
@@ -0,0 +1,374 @@
...
+touch secrets/ocm-service.clientId
+touch secrets/ocm-service.clientSecret
+touch secrets/ocm-service.token
+```
+
+Fleet Manager supports authentication against OCM in two different ways:
+* Using an OCM Service Account
+  * To configure Fleet Manager with this authentication mechanism run:
+    ```
+    echo -n "<your-service-account-client-id> > secrets/ocm-service.clientId
```
Comment (JameelB) [inline docs/populating-configuration.md:41]:
```suggestion
    echo -n "<your-service-account-client-id>" > secrets/ocm-service.clientId
```

File: docs/populating-configuration.md [hunk 27]
```diff
@@ -0,0 +1,374 @@
...
+    echo -n "<your-service-account-client-secret> secrets/ocm-service.clientSecret
```
Comment (JameelB) [inline docs/populating-configuration.md:42]:
```suggestion
    echo -n "<your-service-account-client-secret>" > secrets/ocm-service.clientSecret
```

File: docs/populating-configuration.md [hunk 28]
```diff
@@ -0,0 +1,374 @@
...
+    ```
+* Using an OCM offline token
+  * To configure Fleet Manager with this authentication mechanism run:
+    ```
+    make ocm/setup OCM_OFFLINE_TOKEN=<your-retrieved-ocm-offline-token>
+    ```
+
+> NOTE: If both OCM Service Account and OCM offline token credentials
+  are provided the former takes precedence over the later
```
Comment (JameelB) [inline docs/populating-configuration.md:51]:
```suggestion
  are provided the former takes precedence over the latter
```

File: docs/populating-configuration.md [hunk 29]
```diff
@@ -0,0 +1,374 @@
...
+Fleet Manager is able to create two types of Kafka instances:
+* Developer instances
+  * Instances of this type are automatically deleted after 48 hours by default
+    > NOTE: This can be controlled by setting the `lifespanSeconds` attribute
+      in the `config/kafka-instance-types-configuration.yaml` configuration file
+      for developer instances
+  * All authenticated users that don't have AMS quotas defined to create other
+    kafka instance types can request the creation of a Kafka developer instance
+  * There is a limit of one instance per user by default
+    > NOTE This can be controlled by setting the `--max-allowed-developer-instances`
```
Comment (JameelB) [inline docs/populating-configuration.md:68]:
```suggestion
    > NOTE: This can be controlled by setting the `--max-allowed-developer-instances`
```

File: docs/populating-configuration.md (hunk 26 above)
Comment (miguelsorianod) [reply docs/populating-configuration.md:41]: fixed :+1:

File: docs/populating-configuration.md (hunk 27 above)
Comment (miguelsorianod) [reply docs/populating-configuration.md:42]: fixed :+1:

File: docs/populating-configuration.md (hunk 28 above)
Comment (miguelsorianod) [reply docs/populating-configuration.md:51]: fixed :+1:

File: docs/populating-configuration.md (hunk 29 above)
Comment (miguelsorianod) [reply docs/populating-configuration.md:68]: added :+1:

File: docs/interacting-with-fleet-manager.md [hunk 30]
```diff
@@ -0,0 +1,99 @@
...
+[ocm cli](https://github.com/openshift-online/ocm-cli/releases)
+and retrieve OCM tokens using it. To do so:
+1. Login to your desired OCM environment via web console using your Red Hat
+   account credentials. For example, for the OCM production environment, go to
+   https://console.redhat.com and login.
+1. Get your OCM offline token by going to https://console.redhat.com/openshift/token
+1. Login to your desired OCM environment through the OCM CLI by providing the
+   OCM offline token and environment information:
+   ```
+   ocm login --token <ocm-offline-token> --url <ocm-api-url>
```
Comment (pb82) [inline docs/interacting-with-fleet-manager.md:27]: this can also be copied from console.redhat.com/openshift/token, should we mention it too?

Comment (miguelsorianod) [reply docs/interacting-with-fleet-manager.md:27]: How to obtain the token is already mentioned in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1532/files#diff-42d65bdb3bcf515e8007d9e838cdf0a0ba480e0917b80f1127ddb5be03e5a283R23.
If you are asking about that "ocm login ..." line being able to be coped from that link I don't think that's needed.

**Review by JameelB — APPROVED** [review]
Great work on this @miguelsorianod!
